vnext/ReactUWP/Views/ScrollViewManager.cpp

```diff
@@ -28,6 +15,268 @@ enum class ScrollViewCommands
+    {
+      const auto [valid, showsHorizontalScrollIndicator] = getPropertyAndValidity(propertyValue, true);
+      if (valid)
+      {
+        scrollViewer.HorizontalScrollBarVisibility(showsHorizontalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "showsVerticalScrollIndicator")
+    {
+      const auto [valid, showsVerticalScrollIndicator] = getPropertyAndValidity(propertyValue, true);
+      if (valid)
+      {
+        scrollViewer.VerticalScrollBarVisibility(showsVerticalScrollIndicator ? winrt::ScrollBarVisibility::Visible : winrt::ScrollBarVisibility::Hidden);
+      }
+    }
+    else if (propertyName == "minimumZoomScale")
+    {
+      const auto [valid, minimumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MinZoomFactor(static_cast<float>(minimumZoomScale));
+      }
+    }
+    else if (propertyName == "maximumZoomScale")
+    {
+      const auto [valid, maximumZoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.MaxZoomFactor(static_cast<float>(maximumZoomScale));
+      }
+    }
+    else if (propertyName == "zoomScale")
+    {
+      const auto [valid, zoomScale] = getPropertyAndValidity(propertyValue, 1.0);
+      if (valid)
+      {
+        scrollViewer.ChangeView(nullptr, nullptr, static_cast<float>(zoomScale));
+      }
+    }
+    else if (propertyName == "snapToInterval")
+    {
+      const auto [valid, snapToInterval] = getPropertyAndValidity(propertyValue, 0.0);
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToInterval(static_cast<float>(snapToInterval));
+      }
+    }
+    else if (propertyName == "snapToOffsets")
+    {
+      const auto [valid, snapToOffsets] = getPropertyAndValidity(propertyValue, winrt::single_threaded_vector<float>());
+      if (valid)
+      {
+        ScrollViewUWPImplementation(scrollViewer).SnapToOffsets(snapToOffsets.GetView());
+      }
+    }
+    else if (propertyName == "snapToAlignment")
+    {
+      const auto [valid, snapToAlignment] = getPropertyAndValidity(propertyValue, winrt::SnapPointsAlignment::Near);
+      if (valid)
+      {
+        scrollViewer.HorizontalSnapPointsAlignment(snapToAlignment);
+        scrollViewer.VerticalSnapPointsAlignment(snapToAlignment);
+      }
+    }
+  }
+
+  Super::updateProperties(std::move(reactDiffMap));
+  m_updating = false;
+}
+
+template <typename T>
+std::tuple<bool, T> ScrollViewShadowNode::getPropertyAndValidity(folly::dynamic propertyValue, T defaultValue)
+{
+  if (propertyValue.isNull())
+  {
+    return std::make_tuple(true, defaultValue);
+  }
+
+  if constexpr (std::is_same<T, bool>())
+  {
+    if (propertyValue.isBool())
+    {
+      return std::make_tuple(true, propertyValue.asBool());
+    }
+  }
+
+  if constexpr (std::is_same<T, double>())
```

Comment:
>constexpr (std::is_same<T, double>() [](start = 5, length = 36)

should this allow double / float / int types not just double?

Reply:
For ScrollView the default value is a double, which means the template instantiation will always work so for this control we don't need to allow other types. if we made this more generic we would.